index.html

```diff
@@ -91,43 +95,65 @@ <h2>
         Badge model
       </h2>
       <p>
-        There is a single global badge associated with each <a data-cite=
-        "appmanifest#installable-web-applications">web application</a>. At any time, the application's badge
-        is set to one of the following values:
+        A <dfn>badge</dfn> may be one of the following values:
       </p>
       <ul>
-        <li>The special value <a>nothing</a>.
+        <li>The special value <dfn>nothing</dfn>.
```

Comment:
I think I understand what you mean. The reason I moved it up here is because the previous dfn point was down in the Badge Display section, which is explaining how the values should be used, not defining them. But I see the problem here: I wasn't actually saying what they mean at the (new) definition point.

I've fixed it by adding text up here to explain what "nothing" and "flag" mean. This is the correct place to define them.

Think about it as an algebraic data type / tagged union in whatever your favourite language is (mine's Haskell so let's go):

```haskell
data Badge
  = Nothing     -- No badge set
  | Flag        -- Badge set, but with no data
  | Number Int  -- Badge set to a number
```

This section is essentially the English/spec translation of that data structure. It serves as the definition point for the terms "badge", "nothing" and "flag". So all I needed to do was add comments explaining actually what "nothing" and "flag" mean.